utils/src/lib.rs

```diff
@@ -0,0 +1,77 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2018-2020 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#![warn(missing_docs)]
+
+//! Metadata utilities.
+//!
+//! This crate allows you to parse the latest metadata and use that knowledge
+//! to interact with the chain.
+
+use frame_metadata::RuntimeMetadata;
+
+pub mod error;
+
+/// Latest metadata utilities.
+pub struct Metadata {
```

Comment:
Hi! sorry for meddling. I was doing some metadata parsing work and saw how bad it is to deal with the `RuntimeMetadata` so I added a tiny [util](https://github.com/valibre-org/sube/blob/main/src/meta_ext.rs) for my use case(that could be expanded). My feedback based on my approach is if instead of adding a meaning less wrapping struct what about using an extension trait that you bring to scope and adds the new functionality to the existing `RuntimeMetadata`

Reply:
Hey! Thanks for chiming in. My goal was to actually never expose the actual Metadata types to end user, cause they might change in the future (i.e. changing version V12 -> V13), or may even change during the runtime of your application.
So the idea was to build a set of completely metadata-agnostic utilities such that we would never break backward compatibility.